### ISSUE TITLE
[Bug]TrueNas Datasets does not delete unavailable Datasets

### ISSUE THREAD

**Issue by krzysiek2788** [tomaae/homeassistant-truenas#31]
## Describe the issue
Hi,
I can see that integration does not deleting old datasets which are no more available in TrueNAS Below is a photo. Because it is not deleting it I have more than 1000 sensors 

## How to reproduce the issue
Juset go into TrueNAS Datasets section

## Expected behavior
It will be good that unavailable datasets will be deleted from the list

## Screenshots
![image](https://user-images.githubusercontent.com/61471407/177483250-ee10377c-1144-4cfd-9cb3-672897982852.png)

## Software versions

 - Home Assistant version: 2022.6.7
 - TrueNAS integration version: v1.1.1
 - TrueNAS version: TrueNAS-SCALE-22.02.1

## Diagnostics data
[config_entry-truenas-bfa43443e83bcd003ff195e0d480d8c7.json.txt](https://github.com/tomaae/homeassistant-truenas/files/9052202/config_entry-truenas-bfa43443e83bcd003ff195e0d480d8c7.json.txt)

**Comment by tomaae** [comment]
This is normal behavior in HA. Unavailable entities that you dont expect to be used again are to be deleted manually.

**Comment by krzysiek2788** [comment]
So there is no way to delete all of them by integration? I think integration users are struglling that as well if theyu are using Snapshots.

**Comment by tomaae** [comment]
snapshots do not create new datasets tho. they are bound to a dataset.
having new datasets created and deleted is not a normal operation.

**Comment by krzysiek2788** [comment]
TrueNasScale is creating strange stuff with Docker applications below examples:
![image](https://user-images.githubusercontent.com/61471407/177541030-eb7d86b2-809a-4a4e-b3bb-f809d5b1de9c.png)
![image](https://user-images.githubusercontent.com/61471407/177541068-e5be6ce2-f366-4501-81fa-ffd473d02481.png)

**Comment by tomaae** [comment]
I use truenas core, with scale only in VM due to this integration.
Can you give me complete example configuration to reproduce it? I will have a look what can be done about it.

**Comment by krzysiek2788** [comment]
Which configuration? Integration is nothing as it has only IP and token. Or trueNas Scale?

**Comment by tomaae** [comment]
scale configuration, so I can replicate it in my development instance

**Comment by krzysiek2788** [comment]
You want backup of my configuration db? Or just photos? I think the dataset protection messed it up with ix_application as in my old configuration it was expandable and I saw all docker folders as datasets look at photos:
![image](https://user-images.githubusercontent.com/61471407/177605904-241dd1d3-63e7-4609-99f1-d86a24859b5e.png)
It might be that something messed up after some update of TrueNAS scale. I just recently moved ix-app to SSD pool and it looks like this now:
![image](https://user-images.githubusercontent.com/61471407/177606060-bc47fd72-d96d-4c13-9a75-395d2136b9fd.png)
I will delete old ix-application dataset and delete all unavailable sensor. But it will be good to figure out how to automatically delete unavailable dataset.
